packages/react-reconciler/src/ReactFiberWorkLoop.old.js

```diff
@@ -1517,45 +1517,47 @@ function performUnitOfWork(unitOfWork: Fiber): Fiber | null {
   unitOfWork.memoizedProps = unitOfWork.pendingProps;
   if (next === null) {
     // If this doesn't spawn new work, complete the current work.
-    next = completeUnitOfWork(unitOfWork);
+    completeUnitOfWork(unitOfWork);
+  } else {
+    workInProgress = next;
   }
 
   ReactCurrentOwner.current = null;
-  return next;
 }
 
-function completeUnitOfWork(unitOfWork: Fiber): Fiber | null {
+function completeUnitOfWork(unitOfWork: Fiber): void {
   // Attempt to complete the current unit of work, then move to the next
   // sibling. If there are no more siblings, return to the parent fiber.
-  workInProgress = unitOfWork;
```

Comment:
This line was unnecessary since it's always the current workInProgress that got completed.